compiler/testData/diagnostics/testsWithStdLib/contracts/smartcasts/throwsEffect.fir.kt

```diff
@@ -26,7 +26,7 @@ fun testSpilling(x: Any?) {
         myAssert(x is String)
         x.length
     }
-    x.<!UNRESOLVED_REFERENCE!>length<!>
+    x<!UNSAFE_CALL!>.<!>length
```

Comment:
This one seems interesting. From the path that passed through `if`, `x is String`. From the synthetic `else`, `x == null` (or `x is Nothing?`). After type `or`ing, now `x is String?`. :\

I updated `isAcceptableForSmartcast` accordingly, but seems not good enough. I expect type `or` also needs to drop `Nothing?` at every step?